### PR TITLE
Update admin redirect

### LINKED ISSUE #2944: User invitation: wrong assessment redirect after accepting invitation
**Describe the bug**
User has been invited to panEuropean Cyprus and after having accepted the invitation it gets redirected to fra 2025 landing page .

**To Reproduce**
1. Go to a country and invite a user NC or C or ANC
2. Copy the invitation link and open a new browser to accept invitation.
3. Click on accept invitation 
4. See user has been redirected to fra 2025

**Expected behavior**
At step 4, user should be redirected to invited assessment / cycle / country page

**Video**

https://github.com/openforis/fra-platform/assets/1138156/23118775-c8b0-43aa-a477-1b060b726bca

### PR DESCRIPTION
**Pull request by sorja**
- [x] Update Admin
- [x] Google Login
- [x] Google Accept Invitation
- [x] Local Login
- [x] Local Accept Invitation
- [x] User redirect under country

### CODE REVIEW

File: src/client/pages/Landing/Landing.tsx [hunk 1]
```diff
@@ -17,16 +17,26 @@ const redirectRoles = [
 ]
 
 const Landing: React.FC = () => {
-  const assessment = useAssessment()
+  const assessments = useAssessments()
+  const defaultAssessment = useAssessment()
+  const defaultCycle = useCycle()
```
Comment (minotogna) [inline src/client/pages/Landing/Landing.tsx]: here I think I would not use those hooks. those hooks will need to be updated.
I would create here, an inner hook useRedirectAssessmentAndCycle. Inside this hook, then you can use useAssessments, and useUser to compute them. what do you think?

File: src/client/pages/Country/hooks/useUserRedirect.ts [hunk 2]
```diff
@@ -4,35 +4,32 @@ import { useNavigate } from 'react-router-dom'
...
-  const assessmentName = assessment?.props.name
-  const cycleName = cycle?.name
-
   useEffect(() => {
-    const personalInfoRequired = Users.isPersonalInfoRequired(user, userLastRole)
+    const personalInfoRequired = Users.isPersonalInfoRequired(user, userRole)
     const isFra = assessmentName === AssessmentNames.fra
     const shouldRedirectToProfile = Boolean(
-      isAppInitialized && personalInfoRequired && assessment && cycle && navigate && isFra
+      isAppInitialized && personalInfoRequired && assessmentName && cycle && navigate && isFra
```
Comment (minotogna) [inline src/client/pages/Country/hooks/useUserRedirect.ts]: in country page, app is always initialised and assessment and cycle are always available. I would simplify the condition. what do you think?

File: src/client/pages/Landing/Landing.tsx [hunk 3]
```diff
@@ -2,12 +2,13 @@ import React from 'react'
...
 
 import { Routes } from 'meta/routes'
-import { RoleName } from 'meta/user'
+import { RoleName, Users } from 'meta/user'
 import { UserRoles } from 'meta/user/userRoles'
 
-import { useAssessment, useCycle } from 'client/store/assessment'
 import { useUser } from 'client/store/user'
 
+import { useRedirectAssessmentAndCycle } from './hooks/UseRedirectAssessmentAndCycle'
```
Comment (minotogna) [inline src/client/pages/Landing/Landing.tsx]: file name lower case

File: src/client/pages/Landing/hooks/UseRedirectAssessmentAndCycle.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { UserRoles } from 'meta/user/userRoles'
+
+import { useAppSelector } from 'client/store'
+import { useAssessments } from 'client/store/assessment'
+import { useUser } from 'client/store/user'
+
+export const useRedirectAssessmentAndCycle = () => {
+  const assessments = useAssessments()
+  const { defaultAssessmentId } = useAppSelector((state) => state.assessment.settings)
```
Comment (minotogna) [inline src/client/pages/Landing/hooks/UseRedirectAssessmentAndCycle.tsx]: don't we have a selector for this? or better to create a hook in the store useSettings

File: src/client/pages/Landing/Landing.tsx [hunk 5]
```diff
@@ -17,16 +18,20 @@ const redirectRoles = [
...
   const user = useUser()
-  const userLastRole = UserRoles.getLastRole({ assessment, user })
-  const cycle = useCycle(userLastRole?.cycleUuid)
+  const userLastRole = UserRoles.getLastRole({ user })
 
   const urlParams = { assessmentName: assessment.props.name, cycleName: cycle.name }
+
+  if (Users.isAdministrator(user)) {
+    urlParams.cycleName = assessment.cycles.at(0).name
+  }
```
Comment (minotogna) [inline src/client/pages/Landing/Landing.tsx]: this should be added to the useRedirectAssessmentAndCycle

File: src/client/pages/Landing/hooks/useRedirectAssessmentAndCycle.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+  const userLastRole = UserRoles.getLastRole({ user })
+  const assessment = assessments.find(
+    (assessment) => Number(assessment.id) === Number(userLastRole?.assessmentId ?? defaultAssessmentId)
+  )
+
+  const isAdmin = Users.isAdministrator(user)
+
+  const cycle = isAdmin
+    ? assessment.cycles.at(0)
```
Comment (minotogna) [inline src/client/pages/Landing/hooks/useRedirectAssessmentAndCycle.tsx]: Does admin get redirected to last cycle ?

Comment (sorja) [reply src/client/pages/Landing/hooks/useRedirectAssessmentAndCycle.tsx]: yes

login with admin results:
http://localhost:9000/assessments/fra/2025

Comment (minotogna) [reply src/client/pages/Landing/hooks/useRedirectAssessmentAndCycle.tsx]: How do you guarantee the item at index 0 is the last cycle ? Did I miss the sorting ?

**Review by minotogna — APPROVED** [review]

**Pull request merged**